atlp-rwanda/ecommerce-app-legends-bn
# Ch setup database

## Postgres  database setup
##### set up connection between node js application and postgres data base so that app can access and send data to postgress database

### What does this PR do?
This has functions and methods to set up connection  [ecommerce-app-legends-bn](https://github.com/atlp-rwanda/ecommerce-app-legends-bn) App with postgres database during development phase.

#### what are the nature of PL contents?
---config folder which has js file with db connection configurations
---models folder which has User model created for functionality testing as well index.js which has db connection with 
     models
---migration folder which store table structures to be migrated to the database
---scripts to run both create migration events as well as make: db migrate
---connection folder which has folder to notify the the status of connection from the node js console

N.B : migrations and models folder are still in progress as new and more migrations and modls will b created

### Description of Tasks completed
----database connection set-up successfully
----first model created successfully
----first migration created and migrated successfully
--- make model and create migration as well as migrate scripts added and tested
### how to test this?
clone repo , branch git clone https://github.com/atlp-rwanda/ecommerce-app-legends-bn.git
-`run npm i`
-have `postgres database` installed 
-create a database with `ecommerce_database `as its name
-under config folder replace your postgres credential  in `username and password keys `if you already have them 
-start the app and check connectivity in development `mode npm run dev `
-navigate to `src `folder via terminal and run `npx sequelize-cli model:generate --name renameThis --attributes firstName:string,lastName:string,email:string`  in terminal to make migation  to create new migration
-rename created migration file with your desired file
navigate to `src `folder via terminal and run`npx sequelize-cli db:migrate` to create new table in database
<img width="535" alt="Screen Shot 2023-02-28 at 17 17 59" src="https://user-images.githubusercontent.com/103331579/221897086-da6b95d1-0307-4e44-a900-c2bebc76773c.png">
<img width="680" alt="Screen Shot 2023-02-28 at 17 21 24" src="https://user-images.githubusercontent.com/103331579/221898025-fc598339-1173-4a42-8ee5-cb4ab57b9ef3.png">
